src/libsyntax/lib.rs

```diff
@@ -82,21 +82,23 @@ pub struct Globals {
 }
 
 impl Globals {
-    fn new() -> Globals {
+    /// See documentation on `syntax_pos::Globals` for information on the slice.
+    /// It is solely forwareded to `syntax_pos::Globals::new`
+    fn new(driver_symbols: &[&str]) -> Globals {
         Globals {
             // We have no idea how many attributes their will be, so just
             // initiate the vectors with 0 bits. We'll grow them as necessary.
             used_attrs: Lock::new(GrowableBitSet::new_empty()),
             known_attrs: Lock::new(GrowableBitSet::new_empty()),
-            syntax_pos_globals: syntax_pos::Globals::new(),
+            syntax_pos_globals: syntax_pos::Globals::new(driver_symbols),
         }
     }
 }
 
-pub fn with_globals<F, R>(f: F) -> R
+pub fn with_globals<F, R>(driver_symbols: &[&str], f: F) -> R
```

Comment:
`with_globals` could probably keep its interface and delegate to a new `with_globals_xxx` function taking `&[&str]`.

Too bad Rust still doesn't have default fn parameters to write `fn foo(driver_symbols: &[&str] = &[])`.

Reply:
I was considering that, but it seemed too error prone (since it's too easy to forget to do somewhere).